packages/labs/ssr/src/lib/module-loader.ts

```diff
@@ -8,11 +8,9 @@ import * as path from 'path';
 import {promises as fs} from 'fs';
 import {URL, fileURLToPath, pathToFileURL} from 'url';
 import * as vm from 'vm';
-import resolveAsync from 'resolve';
+import enhancedResolve from 'enhanced-resolve';
```

Comment:
Sorry just saw this; I had needed to add support for export conditions on some of my analyzer exploration work this summer, and `resolve.exports` just dropped in nicely into the `pathFilter` arg of our existing `resolve`: https://github.com/lit/lit/compare/analyzer-pull/#diff-e8a74852b15cd69316ce54daa6fde47bc03b4a8eee3d643bc45546106e65a9c6

Haven't looked deeper into the pros/cons of either approach, but seemed to "just work".

Reply:
That would work but keep in mind it wouldn't solve what al is talking about afaict. 

Both resolve and enhanced-resolve are shipped as commonjs, so with either of these solutions we end up with a commonjs dependency. Though it's already that way so maybe a problem for another pr. 

In that case though I would be tempted to rely on one package for the resolution rather than two.

If we built our own filesystem traversal so are could drop the resolve package, I'd choose resolve.exports. 

If resolve or enhanced-resolve started shipping esm, I'd avoid duplication and just use the one package

Reply:
`resolve.exports` does already ship esm: https://unpkg.com/browse/resolve.exports@1.1.0/dist/index.mjs

Reply:
Oh, maybe I misunderstood; looks like you need to pass a base `resolve` to `resolve.exports`, so it's literally the `resolve` package you are saying is commonjs.

Reply:
yes exactly

`resolve` and `enhanced-resolve` are both commonjs.

`resolve.exports` doesn't implement path resolution, it just implements understanding of the package manifest.

so even if we use it, we need to still use one of the other 2 packages to do the actual filesystem traversal